.github/workflows/build.yaml

```diff
@@ -0,0 +1,279 @@
+        echo "effective_version=${effective_version}" >> "${GITHUB_OUTPUT}"
+        echo "repo_version=${src_version}" >> "${GITHUB_OUTPUT}"
+
+  build:
+    runs-on: ubuntu-latest
+    permissions:
+      id-token: write
+    needs:
+      - params
+      - version
+    strategy:
+      matrix:
+        image:
+          - name: concourse-worker
+            dockerfile: concourse-worker.dockerfile
+            oci_platforms: ${{ needs.params.outputs.oci_platforms }}
+            # todo: handle build-args
+            labels: |
+              - name: gardener.cloud/cve-categorisation
+                value:
+                  network_exposure: public
+                  authentication_enforced: true
+                  user_interaction: gardener-operator
+                  confidentiality_requirement: high
+                  integrity_requirement: high
+                  availability_requirement: high
+          - name: concourse-web
+            dockerfile: concourse-web.dockerfile
+            oci_platforms: linux/amd64
+            labels: |
+              - name: gardener.cloud/cve-categorisation
+                value:
+                  network_exposure: public
+                  authentication_enforced: true
+                  user_interaction: gardener-operator
+                  confidentiality_requirement: high
+                  integrity_requirement: high
+                  availability_requirement: high
+          - name: concourse-resource-git
+            dockerfile: concourse-resource-git.dockerfile
+            oci_platforms: ${{ needs.params.outputs.oci_platforms }}
+            labels: |
+              - name: gardener.cloud/cve-categorisation
+                value:
+                  network_exposure: protected
+                  authentication_enforced: false
+                  user_interaction: gardener-operator
+                  confidentiality_requirement: low
+                  integrity_requirement: high
+                  availability_requirement: low
+          - name: concourse-resource-time
+            dockerfile: concourse-resource-time.dockerfile
+            oci_platforms: ${{ needs.params.outputs.oci_platforms }}
+            labels: |
+              - name: gardener.cloud/cve-categorisation
+                value:
+                  network_exposure: protected
+                  authentication_enforced: false
+                  user_interaction: gardener-operator
+                  confidentiality_requirement: low
+                  integrity_requirement: high
+                  availability_requirement: low
+    outputs:
+      strategy: ${{ toJSON(strategy) }}
+
+    steps:
+      - uses: actions/checkout@v4
+      - name: prebuild-hook
+        if: ${{ matrix.image.name == 'concourse-worker' }}
+        run: |
+          ./prebuild-registry-resource.sh
+      - name: Build OCI Image
+        uses: gardener/cc-utils/.github/actions/ocm-oci-build@gha-matrix-workaround
```

Comment:
I think we should adjust this as soon as the corresponding PR is merged.

Reply:
of course (if we do not, I think this will even fail)